packages/blocks/src/api/registration.js

```diff
@@ -203,10 +203,6 @@ export function registerBlockType( name, settings ) {
 		console.error( 'The "edit" property must be a valid function.' );
 		return;
 	}
-	if ( ! ( 'category' in settings ) ) {
-		console.error( 'The block "' + name + '" must have a category.' );
-		return;
-	}
 	if (
 		'category' in settings &&
 		! some( select( 'core/blocks' ).getCategories(), {
```

Comment:
I guess to support the use-case described at https://github.com/WordPress/gutenberg/pull/19279#issuecomment-625180848, we probably want to make sure that if a block type attempts to register with a category which doesn't actually exist, it should be treated the same as if it were not assigned a category at all. Rather than as it exists here currently, where an error will occur.